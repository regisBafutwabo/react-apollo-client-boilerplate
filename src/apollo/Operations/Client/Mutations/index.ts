import { userStateResolvers } from "./user";
import { i18nResolvers } from "./i18n";
import { Resolvers } from "@apollo/client";

export const LocalResolvers: Resolvers[] = [i18nResolvers, userStateResolvers];
