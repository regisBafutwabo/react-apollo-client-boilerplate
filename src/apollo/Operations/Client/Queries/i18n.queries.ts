import { gql } from "@apollo/client";

export const getLanguage = gql`
    query GetLanguage {
        i18n @client {
            lng
        }
    }
`;

export const updateLanguage = gql`
    mutation UpdateLanguage($language: String!) {
        updateLanguage(language: $language) @client
    }
`;
