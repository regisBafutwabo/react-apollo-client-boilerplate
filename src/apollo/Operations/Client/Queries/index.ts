export * from "./i18n.queries";
export * from "./user.queries";
