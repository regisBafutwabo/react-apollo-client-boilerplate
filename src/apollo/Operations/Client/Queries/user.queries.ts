import { gql } from "@apollo/client";

export const getLocalUser = gql`
    query LocalUser {
        localUser @client {
            isLoggedIn
        }
    }
`;

export const updateLocalUser = gql`
    mutation UpdateLocalUser($isLoggedIn: Boolean!) {
        UpdateLocalUser(isLoggedIn: $isLoggedIn) @client
    }
`;

export const localUserSignOut = gql`
    mutation SignOut {
        signOut @client
    }
`;
