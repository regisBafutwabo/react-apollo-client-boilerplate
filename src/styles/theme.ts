import { DefaultTheme } from "styled-components";

const theme: DefaultTheme = {
    colors: {
        primary: "#282c34",
        secondary: "#61dafb",
        border: "#d9d9d9",
        white: "#ffffff",
    },
};

export default theme;
