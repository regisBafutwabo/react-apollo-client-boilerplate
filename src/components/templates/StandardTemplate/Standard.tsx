import React from "react";
import { Header } from "components/organisms";

import { useQuery, useMutation } from "@apollo/client";
import { Container, Body } from "./styles";
import { getLocalUser, updateLocalUser } from "apollo/Operations/Client/Queries";

const StandardTemplate: React.FC = (props: any) => {
    const { children } = props;

    // Check user status query
    const { loading, error, data } = useQuery(getLocalUser);

    // call logout mutation
    const [Logout] = useMutation(updateLocalUser, {
        variables: { isLoggedIn: false },
    });
    // call login mutation
    const [LogIn, { loading: loginLoader }] = useMutation(updateLocalUser, {
        variables: { isLoggedIn: true },
    });

    return (
        <Container>
            <Header
                Login={LogIn}
                Logout={Logout}
                loading={loading || loginLoader}
                error={error}
                data={data}
            />
            <Body>{children}</Body>
        </Container>
    );
};

export default StandardTemplate;
