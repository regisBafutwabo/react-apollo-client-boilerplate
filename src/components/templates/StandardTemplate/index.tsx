/* eslint-disable no-console */
import React, { ReactNode } from "react";

import { getLocalUser } from "apollo/Operations/Client/Queries";
import { Header } from "components/organisms";
import { AuthService } from "services";

import { useQuery } from "@apollo/client";

import { Body, Container } from "./styles";

const StandardTemplate = ({ children }: { children: ReactNode }) => {
    const { loading, error, data } = useQuery(getLocalUser);

    const login = async () => {
        try {
            await AuthService.login();
        } catch (err) {
            console.error(err);
        }
    };

    const logout = async () => {
        try {
            await AuthService.logout();
        } catch (err) {
            console.error(err);
        }
    };

    return (
        <Container>
            <Header
                Login={login}
                Logout={logout}
                loading={loading}
                error={error}
                data={data}
            />
            <Body>{children}</Body>
        </Container>
    );
};

export default StandardTemplate;
