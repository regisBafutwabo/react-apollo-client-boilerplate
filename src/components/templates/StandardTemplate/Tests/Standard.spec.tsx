/* eslint-disable import/extensions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

import { ThemeProvider } from "styled-components";
import theme from "styles/theme";
import StandardTemplate from "..";
import { getLocalUser } from "apollo/Operations/Client/Queries";

describe("Standard Template", () => {
    // Mocking useLocalUpdateUserMutation and useLocalUserQuery
    const UserMock = [
        {
            request: {
                query: getLocalUser,
                variables: {
                    isLoggedIn: true,
                },
            },
        },
        {
            request: {
                query: getLocalUser,
            },
            result: {
                data: {
                    localUser: {
                        isLoggedIn: true,
                    },
                },
            },
        },
    ];

    it("should render loading ", async () => {
        const Container = render(
            <MockedProvider addTypename={false} mocks={UserMock}>
                <ThemeProvider theme={theme}>
                    <StandardTemplate />
                </ThemeProvider>
            </MockedProvider>,
        );
        // loading should show loading component
        expect(Container).toMatchSnapshot();
    });

    it("should render `logged In` state", async () => {
        const Container = render(
            <MockedProvider addTypename={false} mocks={UserMock}>
                <ThemeProvider theme={theme}>
                    <StandardTemplate />
                </ThemeProvider>
            </MockedProvider>,
        );

        expect(Container).toMatchSnapshot();
    });
});
