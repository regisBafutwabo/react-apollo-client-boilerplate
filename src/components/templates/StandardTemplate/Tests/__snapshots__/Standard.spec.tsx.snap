// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standard Template should render \`logged In\` state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSgupP jrbaVO"
      >
        <header
          class="sc-hKgILt YjBQa"
        >
          <span
            class="ant-dropdown-trigger"
          >
            <span
              aria-label="translation"
              class="anticon anticon-translation"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="translation"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
                />
                <path
                  d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
                />
              </svg>
            </span>
             
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
          <div
            class="ant-spin ant-spin-spinning ant-spin-show-text"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
            <div
              class="ant-spin-text"
            >
              Loading...
            </div>
          </div>
        </header>
        <div
          class="sc-gKsewC PFLUl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSgupP jrbaVO"
    >
      <header
        class="sc-hKgILt YjBQa"
      >
        <span
          class="ant-dropdown-trigger"
        >
          <span
            aria-label="translation"
            class="anticon anticon-translation"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="translation"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
              />
              <path
                d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
              />
            </svg>
          </span>
           
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
        <div
          class="ant-spin ant-spin-spinning ant-spin-show-text"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
          <div
            class="ant-spin-text"
          >
            Loading...
          </div>
        </div>
      </header>
      <div
        class="sc-gKsewC PFLUl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Standard Template should render loading  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSgupP jrbaVO"
      >
        <header
          class="sc-hKgILt YjBQa"
        >
          <span
            class="ant-dropdown-trigger"
          >
            <span
              aria-label="translation"
              class="anticon anticon-translation"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="translation"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
                />
                <path
                  d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
                />
              </svg>
            </span>
             
            <span
              aria-label="down"
              class="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
          <div
            class="ant-spin ant-spin-spinning ant-spin-show-text"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
            <div
              class="ant-spin-text"
            >
              Loading...
            </div>
          </div>
        </header>
        <div
          class="sc-gKsewC PFLUl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSgupP jrbaVO"
    >
      <header
        class="sc-hKgILt YjBQa"
      >
        <span
          class="ant-dropdown-trigger"
        >
          <span
            aria-label="translation"
            class="anticon anticon-translation"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="translation"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
              />
              <path
                d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
              />
            </svg>
          </span>
           
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
        <div
          class="ant-spin ant-spin-spinning ant-spin-show-text"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
          <div
            class="ant-spin-text"
          >
            Loading...
          </div>
        </div>
      </header>
      <div
        class="sc-gKsewC PFLUl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
